accord-core/src/main/java/accord/coordinate/Propose.java

```diff
@@ -140,17 +151,27 @@ else if (reply.committedExecuteAt != null)
         }
     }
 
+    private boolean isSufficientPartialReply(AcceptReply reply, Id from)
+    {
+        return reply.successful != null && reply.successful.containsAll(sendTo.slice(acceptTracker.topologies().computeRangesForNode(from), Minimal));
+    }
+
     @Override
     public void onFailure(Id from, Throwable failure)
     {
+        // TODO (desired): this is a common pattern, find a way to more fully share it
         if (isDone)
             return;
 
-        // TODO (required): we aren't tracking the specific failure here to report
+        if (failure != null)
+            this.failure = FailureAccumulator.append(this.failure, failure);
+
         if (acceptTracker.recordFailure(from) == Failed)
         {
             isDone = true;
-            callback.accept(null, new Timeout(txnId, route.homeKey()));
+            if (this.failure == null)
+                this.failure = new Timeout(txnId, route.homeKey());
```

Comment:
same question about timeout here: since we are calling `onFailure` with `null` failure from success on nack, is it right to replace failure with timeout here?

Reply:
Yes